NDTensors/src/lib/TypeParameterAccessors/src/position.jl

```diff
@@ -0,0 +1,22 @@
+"""
+Represents the compile-time position of a type parameter.
+"""
+struct Position{x} end
+Position(x) = Position{x}()
+
+"""
+    parameter(type::Type, position::Position)
+
+Get a type parameter of the type `type` at the position `position`.
+"""
+parameter(type::Type, position::Position) = parameter(type, Int(position))
+
+struct TypeParameter{P} end
+TypeParameter(x) = TypeParameter{x}()
+
+Base.Int(p::Position) = parameter(p)
```

Comment:
```suggestion
Base.Int(p::Position) = Int(parameter(p))
```
to make sure it really outputs `Int`.